### ISSUE TITLE
Allow overriding error format

### ISSUE THREAD

**Issue by dfalling** [jpiepkow/accesspass#31]
Currently the error responses from AccessPass are usually strings (eg `unauthorized`).  I'd prefer some sort of JSON response.  Is there a way to override this behavior?  Currently I'm overriding the plugs to do this which feels very heavy-handed.

**Comment by jpiepkow** [comment]
Currently no but I think I can add in a customizable behavior this weekend that should work for this. Can you explain your usecase a little more. It is just wanting to replace "unauthorized" with a standard json return like { error: "unauthorized" }

**Comment by dfalling** [comment]
That would be great!  For me it lets me be more consistent in my unit tests and frontend code...my server always returns JSON (except for a 204) so it's very predictable.

**Comment by dfalling** [comment]
And yes- that's basically all I'm doing:
```elixir
  defp unauthorized(conn, message \\ nil) do
    conn
    |> put_status(:unauthorized)
    |> Phoenix.Controller.render(IdoApiWeb.ErrorView, "401.json", message: message)
    |> halt()
  end
```

**Comment by jpiepkow** [comment]
This should not be resolved in 0.6.1 you can set the auth_error function on accesspass behavior to override. Example:


use AccessPassBehavior
def auth_error(conn) do
        send_resp(conn, 401, "different error message") |> halt()
end
